File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;

import entity.PageResult;
import entity.QueryVo;
import entity.Result;

@RestController
@RequestMapping("/brand")
public class BrandController {
	
	
	@Reference //调用远程服务
	private BrandService brandService;
	
	/**
	 * 查询品牌列表
	 * @return
	 */
	@RequestMapping("/findAll.do")
	public List<TbBrand> findAll() {
		return brandService.findAll();
	}
	
	/**
	 * 分页查询品牌数据列表
	 * page 当前页
	 * rows:每页显示的记录数
	 */
	@RequestMapping("/findPage.do")
	public PageResult<TbBrand> findPage(int page,int rows) {
		return brandService.findPage(page, rows);
	}
	
	/**
	 * 添加品牌
	 */
	@RequestMapping("/add.do")
	//使用@RequestBody来接收json类型的请求参数
	public Result add(@RequestBody TbBrand brand) {
		try {
			//调用远程服务
			brandService.add(brand);
			return new Result(true, "添加成功！");
		} catch (Exception e) {
			// TODO: handle exception
			e.printStackTrace(); //在控制台打印错误信息
			//调用远程服务
			return new Result(false, "添加失败！");
		}	
	}
	
	/**
	 * 根据品牌id查询品牌信息
	 */
	@RequestMapping("/findOne.do")
	public TbBrand findOne(Long id) {
		return brandService.findOne(id);
	}
	/**
	 * 修改品牌信息
	 */
	@RequestMapping("/update.do")
	public Result update(@RequestBody TbBrand brand){
		try {
			brandService.update(brand);
			return new Result(true, "修改成功！"); 
		} catch (Exception e) {
			e.printStackTrace(); //向控制台打印异常信息
			return new Result(false, "修改失败！"); 
		}
	}
	
	
	/**
	 * 根据品牌id列表批量删除品牌信息
	 */
	@RequestMapping("/delete.do")
	public Result delete(Long[] ids) {
		try {
			brandService.delete(ids);
			return new Result(true, "删除成功！"); 
		} catch (Exception e) {
			e.printStackTrace(); //向控制台打印异常信息
			return new Result(false, "删除失败！"); 
		}
	}
	
	/**
	 * 根据条件查询并分页
	 */
	@RequestMapping("/search.do")
	public PageResult<TbBrand> search(@RequestBody QueryVo queryVo) {
		return brandService.findPage(queryVo);
	}
	
	/**
	 * 查询指定的字段封装到map中
	 */
	@RequestMapping("/selectOptionList.do")
	public List<Map<String, String>> selectOptionList(){
		//调用服务
		return brandService.selectOptionList();
	}
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
//定义控制层
app.controller('brandController',function($scope,$controller,brandService){
	$controller('baseController',{$scope:$scope});//继承 
	//查询所有品牌列表
	$scope.findAll = function(){
		brandService.findAll().success(function(response){
			$scope.list=response;
		})
	}
	
	//分页查询品牌数据列表
	$scope.findPage = function(page,rows){
		brandService.findPage(page,rows).success(function(response){
			//返回的response是一个封装了分页信息的entity
			$scope.list = response.rows; //每页显示的数据内容
			//重置total 总记录数
			$scope.paginationConf.totalItems = response.total;
		});
	}
	
	//保存品牌信息
	$scope.save=function(){
		//进行判断，根据是否有id来执行添加或者修改操作
		//var methodName = 'add';
		var object = null;
		
		if($scope.entity.id != null){
			//说明执行的update操作
			object = brandService.update($scope.entity);//更新品牌信息
		}else{
			object = brandService.add($scope.entity); //添加品牌信息
		}	
		object.success(function(response){
			if(response.success){
				$scope.reloadList(); //重新加载页面
			}else{
				alert(response.message);
			}
		});
	}
	//根据品牌id查询品牌信息
	$scope.findOne=function(id){
		brandService.findOne(id).success(function(response){
			$scope.entity = response;
		})
	}
	
	
	
	//根据品牌id列表批量删除品牌信息
	$scope.del=function(){
		brandService.del($scope.selectIds).success(function(response){
			if(response.success){
				//清空ids数组中的id
				$scope.selectIds = [];
				$scope.reloadList(); //重新加载页面
			}else{
				alert(response.message);
			}
		})
	}
	
	 $scope.searchEntity = {}; //初始化品牌信息对象
	
	//根据条件查询品牌列表，并进行分页
	$scope.search = function(page,rows){
		 $scope.searchEntity.page = page;
		 $scope.searchEntity.rows = rows;
		brandService.search($scope.searchEntity).success(function(response){
			$scope.list = response.rows; //页面实现的数据内容
			$scope.paginationConf.totalItems = response.total; //更新总记录数
		});
	}
	
})